prog/analysis.go

```diff
@@ -309,3 +310,91 @@ func encodeFallbackSignal(typ, id, aux int) uint32 {
 func decodeFallbackSignal(s uint32) (typ, id, aux int) {
 	return int(s & 7), int((s >> 3) & fallbackCallMask), int(s >> 16)
 }
+
+type pair struct {
+	addr uint64
+	len uint64
+}
+
+func detectIntersection(ranges[]pair) bool{
+	sort.SliceStable(ranges, func(i, j int) bool {
+		return ranges[i].addr < ranges[j].addr
+	    })
+	for i := 0; i < len(ranges)-1; i++ {
+		if ranges[i+1].addr < ranges[i].addr + ranges[i].len {
+			return true
+		}
+	}
+	return false
+}
+
+func filterArguments(call *Call, pointers map[Arg]uint64, p *Prog) bool{
+	var ranges[]pair
+	ForeachArg(call, func(arg Arg, ctx *ArgCtx) {
+		switch a := arg.(type) {
+			case *PointerArg:
+				if ctx.Base == nil {
+					return
+				}
+				value, ok := pointers[arg]
+				addr := p.Target.PhysicalAddr(ctx.Base) + ctx.Offset
+				addr -= arg.Type().UnitOffset()
+				if ok {
+					ranges = append(ranges, pair{addr, value})
+				}else if a.Res != nil {
+					ranges = append(ranges, pair{addr, a.Res.Size()})
+				}
+			case *GroupArg:
+				addr := p.Target.PhysicalAddr(ctx.Base) + ctx.Offset
+				addr -= arg.Type().UnitOffset()
+				ranges = append(ranges, pair{addr, arg.Size()})
+		}
+
+	})
+	return detectIntersection(ranges)
+}
+
+func getPointer(path []string, fields []Field, call *Call) Arg{
+	elem := path[0]
+	var pointer Arg 
+	for i, buf := range call.Args {
+		if elem != fields[i].Name {
+			continue
+		}
+		//TODO: Check on pointer for invalid cases 
+		pointer = buf
+		/*buf = InnerArg(buf)
+		if buf == nil {
+			dst.Val = 0 // target is an optional pointer
+			return
+		}*/
+		break
+		
+	}
+	return pointer
+}
+
+func DFetchAnalysis(p *Prog) bool{
+	for _, call := range p.Calls {
+		pointers := make(map[Arg]uint64)
+		for _, arg := range call.Args{
+			typ, ok := arg.Type().(*LenType)
+			if !ok {
+				continue
+			}
+			lenArg := arg.(*ConstArg)
+			var pointerArg Arg
+			if typ.Path[0] == SyscallRef {
+				pointerArg = getPointer(typ.Path[1:], call.Meta.Args, call)
+			} else {
+				pointerArg = getPointer(typ.Path, call.Meta.Args, call)
+			}
+			pointers[pointerArg] = lenArg.Val
+		}
+		dfDisable := filterArguments(call, pointers, p)
+		if dfDisable {
+			return dfDisable
```

Comment:
I think `return true` is a bit cleaner and more future-proof (if the condition is going to change)